kong/mlcache.lua

```diff
@@ -0,0 +1,466 @@
+-- vim: st=4 sts=4 sw=4 et:
+
+local cjson      = require "cjson.safe"
+local lrucache   = require "resty.lrucache"
+local resty_lock = require "resty.lock"
+
+
+local now          = ngx.now
+local fmt          = string.format
+local sub          = string.sub
+local find         = string.find
+local huge         = math.huge
+local type         = type
+local pcall        = pcall
+local error        = error
+local shared       = ngx.shared
+local tostring     = tostring
+local tonumber     = tonumber
+local setmetatable = setmetatable
+
+
+local LOCK_KEY_PREFIX         = "lua-resty-mlcache:lock:"
```

Comment:
Yeah, I think that's fine, maybe we'll get to replacing this file with the upstream mlcache lib